File: Search/Linear-Search/c/Alternative-Implementation/linearsearch.c
#include <stdio.h>
#include <assert.h>

int linearSearch(int numList[], int arraySize, int number);

int main(void) {
    int numbers[] = {1,2,3,4,5,6,7,8};
    int findNum = 4;

    assert(linearSearch(numbers, sizeof(numbers), findNum) >= 0);
    printf("Number %d was found\n", findNum);
    findNum = 12;
    assert(linearSearch(numbers, sizeof(numbers), findNum) < 0);
    printf("Number %d not found\n", findNum);
}

int linearSearch(int numList[], int arraySize, int number) {
    for (int i = 0; i < arraySize; i++) {
        if (numList[i] == number) {
            return number;
        }
    }
    return -1;
}

File: Search/Linear-Search/js/linearSearch.js
const assert = require('assert');

let numberList = [1,2,3,4,5,6,7,8];
let number = 4;

assert.strictEqual(linearSearch(number, numberList), 3);
console.log(number + ' found');

number = 12
assert.strictEqual(linearSearch(number, numberList), -1);
console.log(number + ' not found');

// Retrun the index otherwise return -1
function linearSearch(number, numberList) {
    for (let i = 0; i < numberList.length; i++) {
        if (numberList[i] === number ) {
            return i;
        }
    };
    return -1;
}